From e694bd75f601d4796119b622ba2f2cd13ca86ddd Mon Sep 17 00:00:00 2001
From: Fernando Lemos <fernandotcl@gmail.com>
Date: Tue, 15 May 2012 14:28:37 +0000
Subject: statusicon: use absolute delta to find the nearest icon size

Otherwise, if we only look at (pixel_size - stock_size), we miss the
case when the pixel size is smaller than any provided stock size.

https://bugzilla.gnome.org/show_bug.cgi?id=673749
---
diff --git a/gtk/gtkstatusicon.c b/gtk/gtkstatusicon.c
index 8bcb638..efecdad 100644
--- a/gtk/gtkstatusicon.c
+++ b/gtk/gtkstatusicon.c
@@ -28,6 +28,7 @@
 #include "config.h"
 
 #include <string.h>
+#include <stdlib.h>
 
 #include "gtkstatusicon.h"
 
@@ -1328,10 +1328,9 @@ round_pixel_size (GtkWidget *widget,
 
   for (s = GTK_ICON_SIZE_MENU; s <= GTK_ICON_SIZE_DIALOG; s++)
     {
-      if (gtk_icon_size_lookup_for_settings (settings, s, &w, &h) &&
-	  w <= pixel_size && h <= pixel_size)
+      if (gtk_icon_size_lookup_for_settings (settings, s, &w, &h))
 	{
-	  d = MAX (pixel_size - w, pixel_size - h);
+	  d = MAX (abs (pixel_size - w), abs (pixel_size - h));
 	  if (d < dist)
 	    {
 	      dist = d;
--
cgit v0.9.0.2

